BlueprintUI/Sources/Layout/Flow.swift

```diff
@@ -0,0 +1,390 @@
+    public func flowChild(key: AnyHashable? = nil) -> Flow.Child {
+        .init(self, key: key)
+    }
+}
+
+
+extension Flow {
+
+    fileprivate struct Layout: BlueprintUI.Layout {
+
+        typealias Traits = Child.Traits
+
+        var horizontalAlignment: HorizontalAlignment
+        var rowAlignment: RowAlignment
+        var horizontalSpacing: CGFloat
+        var rowSpacing: CGFloat
+
+        static var defaultTraits: Traits = .init()
+
+        // MARK: Caffeinated Layout
+
+        func sizeThatFits(
+            proposal: SizeConstraint,
+            subelements: Subelements,
+            environment: Environment,
+            cache: inout ()
+        ) -> CGSize {
+            size(
+                for: subelements.map { $0.sizeThatFits(_:) },
+                in: proposal
+            )
+        }
+
+        func placeSubelements(
+            in size: CGSize,
+            subelements: Subelements,
+            environment: Environment,
+            cache: inout ()
+        ) {
+            zip(
+                frames(
+                    for: subelements.map { $0.sizeThatFits(_:) },
+                    in: .init(size)
+                ),
+                subelements
+            ).forEach { frame, element in
+                element.place(at: frame.origin, size: frame.size)
+            }
+        }
+
+        typealias ElementSize = (SizeConstraint) -> CGSize
+
+        private func frames(
+            for elements: [ElementSize],
+            in constraint: SizeConstraint
+        ) -> [CGRect] {
+
+            var totalHeight: CGFloat = 0
+            var frames: [CGRect] = []
+            var row = rowLayout(
+                origin: .zero,
+                maxWidth: constraint.maximum.width
+            )
+
+            for element in elements {
+                let elementSize = element(constraint)
+
+                // Note: We always want at least one item per row.
+                if !row.canFitItem(of: elementSize) && row.items.isEmpty == false {
+                    frames += row.itemFrames()
+
+                    totalHeight += row.height + rowSpacing
+
+                    row = rowLayout(
+                        origin: .init(x: 0, y: totalHeight),
+                        maxWidth: constraint.maximum.width
+                    )
+                }
+
+                row.addItem(of: elementSize)
+            }
+
+            return frames + row.itemFrames()
+        }
+
+        private func size(
+            for elements: [ElementSize],
+            in constraint: SizeConstraint
+        ) -> CGSize {
+            frames(
+                for: elements,
+                in: constraint
+            ).reduce(.zero) { reduced, current in
+                CGSize(
+                    width: max(current.maxX, reduced.width),
+                    height: max(reduced.height, current.maxY)
+                )
+            }
+        }
+
+        // MARK: Legacy Layout
+
+        func measure(
```

Comment:
Nit: Consider splitting up the different layout functions into extensions on `Layout`?